tests/wpt/web-platform-tests/html/semantics/forms/the-input-element/type-change-state.html

```diff
@@ -36,26 +36,39 @@
       if (types[i] != types[j]) {
         test(function() {
           var input = document.createElement("input");
+          var expected = "  foo\rbar  ";
           input.type = types[i].type;
           if (types[i].type === "file") {
             assert_throws("INVALID_STATE_ERR", function() {
-              input.value = "  foo\rbar  ";
+              input.value = expected;
             });
             assert_equals(input.value, "");
           } else if (types[j].type === "file") {
-            input.value = "  foo\rbar  ";
+            input.value = expected;
             input.type = types[j].type;  // change state
             assert_equals(input.value, "");
           } else {
-            input.value = "  foo\rbar  ";
+            input.value = expected;
             input.type = types[j].type;  // change state
-            if (types[i].type !== "color" && (types[j].sanitizedValue || types[j].sanitizedValue === "")) {
-              assert_equals(input.value, types[j].sanitizedValue, "input.value should be " + types[j].sanitizedValue + " after change of state");
-            } else if (types[i].sanitizedValue || types[i].sanitizedValue === "") {
-              assert_equals(input.value, types[i].sanitizedValue, "input.value should be " + types[i].sanitizedValue + " after change of state");
-            } else {
-              assert_equals(input.value, "  foo\rbar  ", "input.value should be '  foo\\rbar  ' after change of state");
+
+            // type[i] sanitization
+            if (types[i].sanitizedValue || types[i].sanitizedValue === "") {
+              expected = types[i].sanitizedValue;
             }
+
+            // type[j] sanitization
+            if (types[j].sanitizedValue || types[j].sanitizedValue === "") {
+              if ((expected !== "" && !types[i].default) || types[j].default) {
```

Comment:
Perhaps leave a comment here explaining the usage of the `default` field? I can see that it's trying to check whether the source input type has sanitized the `"  foo\rbar  "` string to another value, but I would hesitate to call this the `default`. Maybe `overridesDestinationType` would fit the description more?

Reply:
`overridesSanitization` matches it purpose a way better, I'll change it.